satpy/readers/mviri_l1b_fiduceo_nc.py

```diff
@@ -459,6 +457,29 @@ def __init__(self, nc):
         """Wrap the given dataset."""
         self.nc = nc
 
+        # remove time before decoding and add again.
+        raw_time = nc["time_ir_wv"]
+        self.nc = self.nc.drop_vars(["time_ir_wv"])
+        self.nc = xr.decode_cf(self.nc)
+        self.nc["time_ir_wv"] = raw_time
+
+        self._fix_duplicate_dimensions(self.nc)
+        self.nc = self._chunk(self.nc)
+
+    def _fix_duplicate_dimensions(self, nc):
+        nc.variables["covariance_spectral_response_function_vis"].dims = ("srf_size_1", "srf_size_2")
+
+    def _chunk(self, nc):
+
+        (chunk_size, chunk_size) = nc.variables["quality_pixel_bitmask"].encoding["chunksizes"]
+        chunks = {
+            "x": chunk_size,
+            "y": chunk_size,
+            "x_ir_wv": chunk_size,
+            "y_ir_wv": chunk_size
+        }
```

Comment:
I'd prefer separate x/y chunksizes

```suggestion
        (chunk_size_y, chunk_size_x) = nc.variables["quality_pixel_bitmask"].encoding["chunksizes"]
        chunks = {
            "x": chunk_size_x,
            "y": chunk_size_y,
            "x_ir_wv": chunk_size_x,
            "y_ir_wv": chunk_size_y
        }
```